service/service.go

```diff
@@ -58,6 +59,14 @@ func New(options ...Option) *ExtProcessor {
 func (svc *ExtProcessor) Process(procsrv extproc.ExternalProcessor_ProcessServer) error {
 	req := filter.NewRequestContext()
 	ctx := procsrv.Context()
+	if len(svc.streamCallbacks) > 0 {
+		defer func() {
+			for _, s := range svc.streamCallbacks {
+				s.OnStreamComplete(req)
+			}
+		}()
+	}
```

Comment:
Still not clear here if we want the order to be `0, 1, ... n`, given we reversed the order on the responses (indicating the first filter should have the last word). I believe envoy does 0 to n [here](https://github.com/envoyproxy/envoy/blob/9a04f4c96372b222cd8366aca9f7ebf6122c653a/source/common/http/filter_manager.h#L726)

Reply:
I also noticed it. I'd suggest keeping the same behavior of http connection manager.